compiler/src/jdk.graal.compiler/src/jdk/graal/compiler/nodes/GraphState.java

```diff
@@ -62,22 +62,22 @@ public final class GraphState {
                     StageFlag.GUARD_LOWERING,
                     StageFlag.MID_TIER_LOWERING,
                     StageFlag.FSA,
-                    StageFlag.BARRIER_ADDITION);
+                    StageFlag.MID_TIER_BARRIER_ADDITION);
     private static final EnumSet<StageFlag> LOW_TIER_MANDATORY_STAGES = EnumSet.of(
                     StageFlag.LOW_TIER_LOWERING,
                     StageFlag.EXPAND_LOGIC,
                     StageFlag.ADDRESS_LOWERING,
                     StageFlag.REMOVE_OPAQUE_VALUES,
-                    StageFlag.FINAL_SCHEDULE);
+                    StageFlag.FINAL_SCHEDULE,
+                    StageFlag.LOW_TIER_BARRIER_ADDITION);
     private static final EnumSet<StageFlag> ENTERPRISE_MID_TIER_MANDATORY_STAGES = EnumSet.of(
                     StageFlag.OPTIMISTIC_ALIASING,
                     StageFlag.GUARD_LOWERING,
                     StageFlag.VALUE_PROXY_REMOVAL,
                     StageFlag.SAFEPOINTS_INSERTION,
                     StageFlag.MID_TIER_LOWERING,
                     StageFlag.FSA,
-                    StageFlag.NODE_VECTORIZATION,
-                    StageFlag.BARRIER_ADDITION);
```

Comment:
I think this should have been changed to `StageFlag.LOW_TIER_BARRIER_ADDITION` instead of being deleted.